packages/engine/tests/simulations/state_behavior_index/integration-test.json

```diff
@@ -0,0 +1,16 @@
+[
+  {
+    "experiment": "test",
+    "expected-outputs": [
+      {
+        "json-state": {
+          "1": [
+            {
+              "valid": true
```

Comment:
Do you mean making a test of the testing framework itself, or rather for every test, including a case that should fail?

Reply:
Somewhat the latter, not necessarily including a failure case for **every** test. But we probably do want to check that things don't work when we expect (or require) them to not work. We have that planned somewhat for checking that context is immutable for example. 

Going forwards we'll want to do things like ensuring agents can't modify another agent's state, etc.

Reply:
You are proposing adding run-fail tests?

Reply:
They might be useful too. Depends how deep we want to go. I was thinking of successful runs still, but just verifying failures within the behavior code itself. (Am cautious to use the words try/catch as this might not be about errors)

Reply:
Is this resolved then?